ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -84,22 +85,29 @@ public class OptimizeService extends IJDBCService implements IOptimizeService {
 
   private final BlockingQueue<TableOptimizeItem> toCommitTables = new ArrayBlockingQueue<>(1000);
 
-  private final ConcurrentHashMap<TableIdentifier, TableOptimizeItem> cachedTables = new ConcurrentHashMap<>();
-  private final Set<TableIdentifier> invalidTables = Collections.synchronizedSet(new HashSet<>());
+  private final ConcurrentHashMap<TableIdentifier, TableOptimizeItem> optimizeTables = new ConcurrentHashMap<>();
+  private final Set<TableIdentifier> unOptimizeTables = Collections.synchronizedSet(new HashSet<>());
 
   private final OptimizeQueueService optimizeQueueService;
   private final IMetaService metaService;
-  private final AmsClient metastoreClient;
+  private final ExecutorService tablesSvc;
   private volatile boolean inited = false;
 
   public OptimizeService() {
     super();
     optimizeQueueService = ServiceContainer.getOptimizeQueueService();
     metaService = ServiceContainer.getMetaService();
-    metastoreClient = ServiceContainer.getTableMetastoreHandler();
+    tablesSvc = getProcessTablesThreadPool();
     init();
   }
 
+  private ExecutorService getProcessTablesThreadPool() {
+    if (tablesSvc == null || tablesSvc.isShutdown() || tablesSvc.isTerminated()) {
+      return Executors.newFixedThreadPool(loadTablePoolSize());
+    }
+    return tablesSvc;
+  }
+
```

Comment:
I think this will bring a memory leak problem, creating new `FixedThreadPool` endlessly but can't shut them down.

Reply:
fix by https://github.com/NetEase/arctic/pull/1292